pkg/controllers/provisioning/v1alpha1/allocation/constraints.go

```diff
@@ -101,6 +105,32 @@ func (c *Constraints) getLabels(provisioner *v1alpha1.Provisioner, pod *v1.Pod)
 	)
 }
 
+func (c *Constraints) getZones(provisioner *v1alpha1.Provisioner, pod *v1.Pod) []string {
+	// Pod may override zone
+	if zone, ok := pod.Spec.NodeSelector[ZoneLabelKey]; ok {
+		return []string{zone}
+	}
+	// Default to provisioner constraints
+	if provisioner.Spec.Zones != nil {
```

Comment:
I think this should be a `if len(provisioner.Spec.Zones) == 0`

Reply:
+1
When I was thinking about this, I thought that [] meant constrained but no options (i.e. always fails). Thinking again, this is probably a useless "feature" and instead should probably semantically mean "unconstrained".